app/src/main/java/org/sopt/sample/src/login/SignInActivity.kt

```diff
@@ -0,0 +1,105 @@
+package org.sopt.sample.src.login
+
+import android.app.Activity
+import android.content.Intent
+import android.content.SharedPreferences
+import android.os.Bundle
+import android.util.Log
+import androidx.activity.result.ActivityResultLauncher
+import androidx.activity.result.contract.ActivityResultContract
+import androidx.activity.result.contract.ActivityResultContracts
+import org.sopt.sample.R
+import org.sopt.sample.config.ApplicationClass
+import org.sopt.sample.config.BaseActivity
+import org.sopt.sample.databinding.ActivitySigninBinding
+import org.sopt.sample.src.HomeActivity
+
+
+class SignInActivity : BaseActivity<ActivitySigninBinding>(ActivitySigninBinding::inflate) {
+    private lateinit var resultLauncher: ActivityResultLauncher<Intent> //회원가입 정보를 받아오기 위한 launcher
+    private var idFromSignup: String? = null
+    private var pwFromSignup: String? = null
+    private var mbtiFromSignup: String? = null
+    private lateinit var editor:SharedPreferences.Editor
+    override fun onCreate(savedInstanceState: Bundle?) {
+
+        super.onCreate(savedInstanceState)
+        editor = ApplicationClass.sSharedPreferences.edit()
+
+        autoLogin()//자동 로그인
+        setResultSignUp() //회원가입 콜백
+        signIn() //로그인
+        signUp() //회원가입
+
+    }
+
+    //Signup Callback
+    private fun setResultSignUp() {
+        resultLauncher =
+            registerForActivityResult(ActivityResultContracts.StartActivityForResult()) { result ->
+                if (result.resultCode == Activity.RESULT_OK) {
+                    idFromSignup = result.data?.getStringExtra("id").toString()
+                    pwFromSignup = result.data?.getStringExtra("pw").toString()
+                    mbtiFromSignup = result.data?.getStringExtra("mbti").toString()
+                    CustomSnackBar(getString(R.string.signin_success_signup)).setAnchorView(binding.signInLoginBtn)
+                        .show()
+                }
+                //회원가입 취소
+                else {
+                    CustomSnackBar(getString(R.string.signin_fail_signup)).setAnchorView(binding.signInLoginBtn)
+                        .show()
+                }
+            }
+    }
+    private fun signIn() {
+        //로그인 버튼 클릭
+        binding.signInLoginBtn.setOnClickListener {
+            //SignUp에서 받아온 id/pw와 일치하는지 체크
+            if (binding.signInIdEt.text.toString() == idFromSignup &&
+                binding.signInPwEt.text.toString() == (pwFromSignup)
+            ) {
+                showCustomToast("로그인에 성공했습니다")
+
+                //로그인 성공 시 id,pw,mbti를 sp에 저장
+                editor.putString("id",idFromSignup)
+                editor.putString("pw",pwFromSignup)
+                editor.putString("mbti",mbtiFromSignup)
+                editor.commit()
```

Comment:
아래와 같이 수신객체(editor)의 프로퍼티를 변경할 때는 apply를 사용해볼 수도 있겠습니다~
```suggestion
                //로그인 성공 시 id,pw,mbti를 sp에 저장
                editor.apply {
                    putString("id",idFromSignup)
                    putString("pw",pwFromSignup)
                    putString("mbti",mbtiFromSignup)
                    commit()
                }
```
이미 알고계신다면 패스하셔도 됩니당 :) ▶️ [kotlin의 범위지정함수](https://medium.com/@limgyumin/%EC%BD%94%ED%8B%80%EB%A6%B0-%EC%9D%98-apply-with-let-also-run-%EC%9D%80-%EC%96%B8%EC%A0%9C-%EC%82%AC%EC%9A%A9%ED%95%98%EB%8A%94%EA%B0%80-4a517292df29)

Reply:
@youngjinc 애매하게 알고 있었는데 정리가 너무 잘 되어있네요~ 감사합니다!